src/customGraphQLValidationRules.js

```diff
@@ -34,32 +39,29 @@ export function RequiredFields(context, options) {
 
   return {
     FragmentDefinition(node) {
-      requiredFields.forEach(field => {
-        const type = context.getType();
+      const type = context.getType();
 
-        if (fieldAvailableOnType(type, field)) {
-          const fieldWasRequested = getFieldWasRequestedOnNode(node, field);
-          if (!fieldWasRequested) {
-            context.reportError(
-              new GraphQLError(
-                `'${field}' field required on 'fragment ${node.name.value} on ${node.typeCondition.name.value}'`,
-                [node]
-              )
-            );
-          }
+      requiredFields.forEach(field => {
+        if (fieldAvailableOnType(type, field) && !fieldIsRequestedOnNode(node, field)) {
+          context.reportError(
+            new GraphQLError(
+              `'${field}' field required on 'fragment ${node.name.value} on ${node.typeCondition.name.value}'`,
+              [node]
+            )
+          );
         }
       });
     },
 
     // Every inline fragment must have the required field specified inside
     // itself or in some parent selection set.
     InlineFragment(node, key, parent, path, ancestors) {
-      requiredFields.forEach(field => {
-        const type = context.getType();
+      const type = context.getType();
 
+      requiredFields.forEach(field => {
         if (fieldAvailableOnType(type, field)) {
           // First, check the selection set on this inline fragment
-          if (node.selectionSet && getFieldWasRequestedOnNode(node, field)) {
+          if (fieldIsRequestedOnNode(node, field)) {
```

Comment:
Here `node.selectionSet` is mandatory since you can't write an inline fragment without a selection set.